### ISSUE TITLE
`yq eval --inplace` should support writing changes back to each individual file

### ISSUE THREAD

**Issue by zx8** [mikefarah/yq#900]
**Please describe your feature request.**

Currently, as stated in the help text, `yq eval--inplace` writes back all changes to the _first_ file in the list. This always struck me as a bit odd and is different to how other tools treat in-place replacements (e.g. `sed -i` changes each individual file)

**Describe the solution you'd like**

It would be great if `--inplace` edited each file in-place, rather than merging all changes into the first file - this behaviour just seems odd. I realise this is a breaking change, so it might be better to just introduce a separate flag for it.

**Describe alternatives you've considered**

The only alternative I can think of is to use a for-loop and edit each individual file with separate `yq` invocations:

```sh
for f in *.yaml; do
  yq eval '.foobar=true' "$f"
done
```

**Comment by mikefarah** [comment]
You can also use `find`:

```
find *.yaml -exec yq e '.foorbar=true' -i {} \;
```

It does make sense for `-i` to update each file for the `eval` command, however not for the `eval-all` command where most often its used to merge multiple files into one.

Given that there's a work around using find and bash loops and probably more, I think I'll leave this for now

**Comment by zx8** [comment]
Completely agree on the `eval` vs. `eval-all` point!

> I think I'll leave this for now

That's a shame – would you be open to a merging PR that implemented this?

**Comment by mikefarah** [comment]
Yeah sure! If it's implemented as a new flag for eval-seq, and the code is clean  and tested :)

**Comment by mikefarah** [comment]
Closing due to inactivity

**Comment by techdragon** [comment]
@mikefarah given your happy to have a PR with this feature, would you consider reopening it so that someone is more likely to find it and consider working on it.

**Comment by mikefarah** [comment]
fair call - yep
